x/nft/client/rest/query.go

```diff
@@ -180,6 +183,40 @@ func queryDenom(cliCtx client.Context, queryRoute string) http.HandlerFunc {
 	}
 }
 
+// nolint: dupl
+func queryDenomByName(cliCtx client.Context, queryRoute string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// nolint: govet
+		cliCtx, ok := rest.ParseQueryHeightOrReturnBadRequest(w, cliCtx, r)
+		if !ok {
+			return
+		}
+
+		denomName := mux.Vars(r)[RestParamDenomName]
+		if err := types.ValidateDenomName(denomName); err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+		}
+
+		params := types.NewQueryDenomByNameParams(denomName)
+		bz, err := cliCtx.LegacyAmino.MarshalJSON(params)
```

Comment:
A question relates to the codec. Is any point we will use `gogoprotobuf ` instead of `amino`?

Reply:
Ideally, clients should directly use gRPC. Rest can be deprecated in the future.